ols/app/models/config.py

```diff
@@ -183,6 +184,13 @@ def __init__(self, data: Optional[dict] = None) -> None:
                 raise InvalidConfigurationError("model name is missing")
             model = ModelConfig(m)
             self.models[m["name"]] = model
+        if self.type == constants.PROVIDER_AZURE_OPENAI:
+            # deployment_name only required when using Azure OpenAI
+            self.deployment_name = data.get("deployment_name", None)
+            if self.deployment_name is None:
+                raise InvalidConfigurationError(
+                    f"deployment_name is required for Azure OpenAI provider {self.name}"
```

Comment:
we need a test that ensures this is enforced, especially so when @m-danilow refactors our validation logic, we've got a test that enforces the expected behavior.

i think that is the last thing needed here.  we're also going to have to work out how to have configs for azureai that specify the right deployment name value, but we can handle that in a followup refactoring of how e2e config is managed.